versions/development/SPEC.md

```diff
@@ -1790,6 +1789,83 @@ workflow foo {
 }
 ```
 
+### Info and Debug Logging
+
+Log and debug statements (`info` and `debug`) allow a workflow author to signal to the engine to log a value at either *info* or *debug* level.
+
+The exact mechanism of displaying logged statements to the user is left to the engine and may well be different for different workflows:
+
+* A command line invocation of a workflow may well 'log' to the console's stdout, for example.
+* On the other hand a server running a workflow might 'log' into a completion report that it provides to the user on job completion. 
+* An engine if free to provide any additional context that it chooses as it processes the log statements (a timestamp or a fully qualified path to the current workflow, for example)
+* An engine may even decide not to expose any logs to the end-user at all, if that makes sense in its current execution mode.
+
+The value following the `log` or `info` can be any expression. It must evaluate to a `String` or `String?` value, or a value which can be coerced into `String` or `String?`
```

Comment:
should there be a discussion about **when** a log/debug statement is evaluated? calls are executed whenever their inputs are defined "enough", is it the same for log? Perhaps provide an example of an out-of order workflow and be explicit about when each thing is executed.